pkg/controller/azvolumeattachment.go

```diff
@@ -82,14 +98,14 @@ func (r *reconcileAzVolumeAttachment) Reconcile(ctx context.Context, request rec
 func (r *reconcileAzVolumeAttachment) HandleAzVolumeAttachmentEvent(ctx context.Context, request reconcile.Request) (reconcile.Result, error) {
 	var azVolumeAttachment v1alpha1.AzVolumeAttachment
 	err := r.client.Get(ctx, request.NamespacedName, &azVolumeAttachment)
-	// if object is not found, it means the object has been deleted. Log the deletion and do not requeue
+	// if object is not found, it means either 1) the object has been deleted. Log the deletion and do not requeue or 2) the object is not AzVolumeAttachment type. return NotFound error
```

Comment:
We should use separate reconciler functions to avoid this ambiguity.

Reply:
My apology. This comment was added when I was experimenting with adding PVC watch to AzVolumeAttachment controller. I should edit the comment here, AzVolumeAttachment controller reconciler only watches for a single object so there shouldn't be such ambiguity.

I will fix the comment to prevent any confusion.